compiler/graindoc/docblock.re

```diff
@@ -420,7 +516,62 @@ let for_type_declaration =
   let name = Format.asprintf("%a", Printtyp.ident, ident);
   let type_sig = Printtyp.string_of_type_declaration(~ident, td);
   let comment =
-    Comments.Doc.ending_on(~lnum=loc.loc_start.pos_lnum - 1, comments);
+    Comments.Doc.ending_on_including_attribute(
+      ~lnum=loc.loc_start.pos_lnum - 1,
+      comments,
+    );
+
+  let extract_compound_type_descrs = (datas, mk_type_descr) => {
+    List.map(
+      ((data, loc: Warnings.loc, id)) => {
+        let comment =
+          Comments.Doc.ending_on(~lnum=loc.loc_start.pos_lnum - 1, comments);
+        switch (comment) {
+        | Some((_, _, [attr, ..._])) =>
+          raise(
+            InvalidAttribute({
+              name: Format.asprintf("%a", Printtyp.ident, id),
+              attr: attr_name(attr),
+            }),
+          )
```

Comment:
Does this need the logic for non-duplicate `@since` that @spotandjake added in the other PR?

Reply:
This would raise the exception if there are any attributes on the docblock at all, so deduplicating isn't necessary here

Reply:
Hm, I'm pretty sure the JSON PR uses the since attribute on each variant so we'll want to support that

Reply:
Hmm ok, should `@deprecated` and `@history` be supported as well?

Reply:
Yeah, I think those would be supported

Reply:
What about record fields? And if so then how would they be displayed in the table?

Reply:
It's probably fine to only support them on variants. When thinking about exhaustive matching, I think about variants more than record fields and you might add variants over time.

Reply:
Hmm, I think it'd be confusing if they were supported on variants but not record fields; for instance if you had a field that changed its type or name or something. What if it's just inserted below the field's description in the table cell? It wouldn't take up much space by default since it'd be collapsed in the `<details>` tag

Reply:
I don't want it in the table. We can open a follow up issue to evaluate for records

Reply:
Merging without attribute support and @spotandjake can remove them from the JSON PR

Reply:
#1998 and #1999